index.html

```diff
@@ -1177,119 +1177,269 @@ <h2 id="relationship-to-permissions-policy">
         permission=] or by some user agent policy.
       </p>
     </section>
-    <section data-cite="webdriver">
+    <section>
       <h2 id="automation">
         Automated testing
       </h2>
       <p>
         For the purposes of user-agent automation and application testing, this document defines
-        the following <a>extension commands</a> for the [[WebDriver]] specification. It is OPTIONAL
-        for a user agent to support <a>extension commands</a> commands.
+        extensions to the [[WebDriver]] and [[WebDriver-BiDi]] specifications.
+        It is OPTIONAL for a user agent to support them.
       </p>
       <pre class='idl'>
         dictionary PermissionSetParameters {
           required PermissionDescriptor descriptor;
           required PermissionState state;
         };
       </pre>
-      <section>
-        <h3 id="set-permission-command">
-          Set Permission
+      <p>To <dfn>set a permission</dfn> given <var>descriptor</var>, and <var>state</var>:
+      <ol class="algorithm">
+        <li>Let |settings| be the [=current settings object=].
+        </li>
+        <li>Let |targets| be a <a>list</a> containing all <a>environment settings objects</a>
+        whose [=environment settings object/origin=] is the <a data-lt="same origin">same</a> as
+        the [=environment settings object/origin=] of |settings|.
+        </li>
+        <li>Let |tasks| be an empty <a>list</a>.
+        </li>
+        <li>For each <a>environment settings object</a> |target| in |targets|:
+          <ol>
+            <li>
+              <a>Queue a task</a> |task| on the [=permissions task source=] of |target|'s
+              [=relevant settings object=]'s [=environment settings object/global object=]'s
+              [=Window/browsing context=] to perform the following step:
+              <ol>
+                <li>Interpret |state| as if it were the
+                result of an invocation of <a>permission state</a> for |descriptor| with the
+                argument |target| made at this moment.
+                </li>
+              </ol>
+            </li>
+            <li>[=list/Append=] |task| to |tasks|.
+            </li>
+          </ol>
+        </li>
+        <li>Wait for all <a>tasks</a> in |tasks| to have executed.
```

Comment:
```suggestion
        <li>Wait for all <a>tasks</a> in |tasks| to have executed and return.
```

Reply:
maybe?

Reply:
Yeah, what's the intent here with waiting?

Reply:
This is pre-existing prose: https://www.w3.org/TR/permissions/#set-permission-command

It seems to me the intent is to just "await" before returning from the method/function, for syncing purposes

Reply:
Ah, now I see the point of adding the `and return`, because it's extracted into an algorithm. Done.